cmd/minikube/cmd/start.go

```diff
@@ -353,7 +355,27 @@ func runStart(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	kubeconfig := node.Start(cc, n, existingAddons, true)
+	kubeconfig, err := node.Start(cc, n, existingAddons, true)
+	if err != nil && viper.GetBool(deleteOnFailure) {
+		out.T(out.Warning, "Cluster {{.name}} failed to start, deleting and trying again.", out.V{"name": cc.Name})
+		// Start failed, delete the cluster and try again
+		profile, err := config.LoadProfile(cc.Name)
+		if err != nil {
+			out.ErrT(out.Meh, `"{{.name}}" profile does not exist, trying anyways.`, out.V{"name": cc.Name})
+		}
+
+		err = deleteProfile(profile)
+		if err != nil {
+			// We failed to delete? That's not good. Error out.
+			exit.WithError("deleting cluster failed", err)
```

Comment:
I feel like there is nothing to lose by trying `node.Start` again at this point. Can this be an `out.Warning` instead of an exit?

Reply:
yep, done